src/core/server/saved_objects/service/lib/delete_index_templates.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ElasticsearchClient } from '../../../elasticsearch';
+import type { Logger } from '../../../logging';
+
+interface DeleteIndexTemplatesOptions {
+  client: ElasticsearchClient;
+  log: Logger;
+}
+
+/**
+ * Deletes old index templates that were used in 6.x as those will no longer be supported in ES 8.
+ */
+export const deleteIndexTemplates = async ({ client, log }: DeleteIndexTemplatesOptions) => {
+  const { body: templates } = await client.cat.templates({
```

Comment:
there was recently an email thread about not using `cat` API's in production code since they don't have good backwards compatibility guarantees and often have bad performance. Although it's not documented they were actually intended as debugging API's only.

I think `GET /_index_template/kibana_index_template*` will give the same result.

Reply:
done in https://github.com/elastic/kibana/pull/119150/commits/273ccdf82f716176a36a37a3a076a2d0bb67d8c7